src/app/actions/admin/products.ts

```diff
@@ -3,22 +3,37 @@ import { AvailabilityStatus, Category, returnPolicy } from '@/types/product';
 import { z } from 'zod';
 import { db, collections } from '@/utils/firebase';
 import { doc, setDoc } from "firebase/firestore";
+import { put } from '@vercel/blob';
+
 
 export const productSchema = z.object({
-  title: z.string().min(3, "Title must be at least 3 characters").max(100),
-  description: z.string().min(50, "Description must be at least 50 characters").max(500),
+  title: z.string().min(3).max(100),
+  description: z.string().min(50).max(500),
   category: z.nativeEnum(Category),
-  price: z.number().min(0, "Price cannot be negative"),
-  stock: z.number().min(0, "Stock cannot be negative"),
-  brand: z.string().optional(),
+  price: z.number().min(0),
+  stock: z.number().min(0),
+  brand: z.string().min(1),
   availabilityStatus: z.nativeEnum(AvailabilityStatus),
-  returnPolicy: z.nativeEnum(returnPolicy)
+  returnPolicy: z.nativeEnum(returnPolicy),
+  tags: z.array(z.string()).min(1),
+  sku: z.string().min(1).max(50),
+  weight: z.number().min(1),
+  warrantyInformation: z.string().min(1),
+  shippingInformation: z.string().min(1),
+  dimensions: z.object({
+    width: z.number().min(1),
+    height: z.number().min(1),
+    depth: z.number().min(1),
+  }),
 });
+
+
 export async function addNewProductAction(
   currentState: NewProductFormState,
   formData: FormData
 ): Promise<NewProductFormState> {
 
+  
   const rawData = {
```

Comment:
The form data to raw data convertion is very wordy, make sure that you have a function for that and that you re-use it both here and in the update product action.